sympy/core/tests/test_function.py

```diff
@@ -23,6 +23,25 @@ def test_f_expand_complex():
     assert exp(z).expand(complex=True) == cos(im(z))*exp(re(z)) + \
         I*sin(im(z))*exp(re(z))
 
+def test_unicode_name_support():
+    Function(u('k')) # check if unicoded function names can be created
+
+    # check if unicoded function is same as its ascii counterpart
+    assert Function('k') == Function(u('k'))
+
+def test_greek_name_support():
+    Function('α')
+    Function(u('α'))
+    Function('α') == Function(u('α'))
+
+    Function('αβ')
+    Function(u('αβ'))
+    Function('αβ') == Function(u('αβ'))
+
+    Function('aαβ')
+    Function(u('aαβ'))
+    Function('aαβ') == Function(u('aαβ'))
+
```

Comment:
The thing was that I tried around three implementations to solve this problem.
* The first one solved the case for a single character length function names. But if you added multiple greek characters., they failed.
* Similarly a combination of ASCII and non-ASCII failed my second implementation.

In order to document all these I added them in a test case.
They are basically special cases.

I will work on adding the documentation for them.